### PR TITLE
docs: document some useful git hooks

### PR DESCRIPTION
**Pull request by bstrie**
These are the hooks I use to keep myself from accidentally
wasting CI time and thought others might benefit from them.
Also fix up some old repo links.

### CODE REVIEW

File: docs/CONTRIBUTING.md [hunk 1]
```diff
@@ -30,22 +30,32 @@ Before diving in to Interledger.rs, you may find it helpful to familiarize yours
...
 - Make sure to commit using `-s` or `--signoff` option like `git cz -s`.
     - `cz` means using the [commitizen](https://github.com/commitizen/cz-cli) explained below.
     - Why we use the option is explained later in the [Signing-off](#Signing-off) section.
 - Make sure that you committed using the [commitizen](https://github.com/commitizen/cz-cli) format (commit messages should start with `feat:`, `docs:`, `refactor:`, `chore:`, etc). PRs may contain multiple commits but they should generally be squashed into one or a small number of complete change sets (for example, a feature followed by multiple refactors and another commit to add tests and docs should be combined into a single commit for that feature).
-- Make pull requests against `master` branch of this repository (emschwartz/interledger-rs) from your repository.
+    - If you would like to make your local setup reject improperly formatted commit headers, you can add the following code to `interledger-rs/.git/hooks/commit-msg`:
+```bash
+if head -n 1 "$1" | grep -vqE "(?:^(?:feat|fix|docs|style|refactor|perf|test|chore)(:?\(\S{1,30}\))?:[ ].{5,100}$)"; then
```
Comment (emschwartz) [inline docs/CONTRIBUTING.md]: Note https://github.com/interledger-rs/interledger-rs/commit/2589b04b669c145ba0ad53878fdfa702f6268470 slightly changed the regex

Comment (dora-gt) [reply docs/CONTRIBUTING.md]: image

Don't we need to add these?

Comment (dora-gt) [reply docs/CONTRIBUTING.md]: Also, I'm not sure this is intended or not.
image

```
# Commit Message Format
<type>(<scope>): <subject>
<BLANK LINE>
<body>
<BLANK LINE>
<footer>
```

It looked a bit weird for me (Sorry if I'm missing something).
image

Comment (emschwartz) [reply docs/CONTRIBUTING.md]: > Don't we need to add these?

Yes

> Also, I'm not sure this is intended or not.

The `:?` just means it's a "non-capturing group". I don't think that makes a difference in this case so we can probably get rid of those (also in the very first group)

Comment (dora-gt) [reply docs/CONTRIBUTING.md]: Isn't it `?:`?
It is `:?`

Comment (dora-gt) [reply docs/CONTRIBUTING.md]: and... the answer reminds me of the typical English question.

1. Yes, we need to add
1. Yes, we don't need to add

In Japan, it is said that 1 is correct, which do you mean? 😅

Comment (emschwartz) [reply docs/CONTRIBUTING.md]: I think we can change the regex to:
`(^(feat|fix|docs|style|refactor|perf|test|chore|ci)(\(.{1,30}\))?:[ ].{5,100}$)(\n(\n{1,2}^.{1,100}$)+)?`

Comment (emschwartz) [reply docs/CONTRIBUTING.md]: I have definitely seen and used `ci` though not `build`

File: docs/CONTRIBUTING.md [hunk 2]
```diff
@@ -30,22 +30,29 @@ Before diving in to Interledger.rs, you may find it helpful to familiarize yours
...
 - Make sure to commit using `-s` or `--signoff` option like `git cz -s`.
     - `cz` means using the [commitizen](https://github.com/commitizen/cz-cli) explained below.
     - Why we use the option is explained later in the [Signing-off](#Signing-off) section.
 - Make sure that you committed using the [commitizen](https://github.com/commitizen/cz-cli) format (commit messages should start with `feat:`, `docs:`, `refactor:`, `chore:`, etc). PRs may contain multiple commits but they should generally be squashed into one or a small number of complete change sets (for example, a feature followed by multiple refactors and another commit to add tests and docs should be combined into a single commit for that feature).
-- Make pull requests against `master` branch of this repository (emschwartz/interledger-rs) from your repository.
+    - If you would like to make your local setup reject improperly formatted commit headers, you can add the following code to `interledger-rs/.git/hooks/commit-msg`:
+```bash
+if head -n 1 "$1" | grep -vqE "^(feat|fix|docs|style|refactor|perf|test|chore|ci|build)(\(.{1,30}\))?:[ ].{5,100}$"; then
```
Comment (emschwartz) [inline docs/CONTRIBUTING.md]: This doesn't exactly match the regex being run on CI right now. Can you make sure they are the same (the one being run on CI should be updated to include `ci` and `build`)

**Comment by bstrie** [comment]
Closing in favor of #309 per our new branching policy.

**Pull request closed without merging**